Integrations/ExtraHop/ExtraHop.py

```diff
@@ -0,0 +1,367 @@
+    active_until = demisto.args().get('active_until')
+    search_type = demisto.args().get('search_type')
+    limit = demisto.args().get('limit')
+    payload = {}
+    if active_from:
+        payload['active_from'] = active_from
+    if active_until:
+        payload['active_until'] = active_until
+    if limit:
+        payload['limit'] = limit
+    payload['search_type'] = search_type
+    res_raw = http_request('GET', 'devices', data=None, payload=payload)
+    res = res_raw.json()
+    return res
+
+
+def format_alerts(alerts):
+    hr = ''
+    ec = {
+        "ExtraHop": {
+            "Alert": []
+        }
+    }
+    for alert in alerts:
+        hr += tableToMarkdown('Found Alert', alert, headerTransform=string_to_table_header, removeNull=True)
+        ec['ExtraHop']['Alert'].append(createContext(alert, keyTransform=string_to_context_key, removeNull=True))
+    if len(alerts) == 0:
+        demisto.results('No results were found')
+    else:
+        demisto.results({
+            'Type': entryTypes['note'],
+            'ContentsFormat': formats['markdown'],
+            'Contents': alerts,
+            'HumanReadable': hr,
+            'EntryContext': ec
+        })
+
+
+def format_device_results(data):
+    hr_table = []
+    ec = {
+        "ExtraHop": {
+            "Device": []
+        }
+    }
+    for device in data:
+        hr = {}
+        if 'id' in device:
+            hr['ID'] = device.get('id')
+        if 'display_name' in device:
+            hr['Display Name'] = device.get('display_name')
+        if 'ipaddr4' in device:
+            hr['IP Address'] = device.get('ipaddr4')
+        if 'macaddr' in device:
+            hr['MAC Address'] = device.get('macaddr')
+        if 'vendor' in device:
+            hr['Vendor'] = device.get('vendor')
+        hr_table.append(hr)
+        ec['ExtraHop']['Device'].append(createContext(device, keyTransform=string_to_context_key, removeNull=True))
+    demisto.results({
+        'Type': entryTypes['note'],
+        'ContentsFormat': formats['markdown'],
+        'Contents': data,
+        'HumanReadable': tableToMarkdown('Devices Found', hr_table),
+        'EntryContext': ec
+    })
+
+
+def whitelist_modify(add, remove):
+    assignments = {}
+    if add:
+        add_items = add.split(',')
+        add_items = list(map(int, add_items))
+        assignments['assign'] = add_items
+    if remove:
+        remove_items = remove.split(',')
+        remove_items = list(map(int, remove_items))
+        assignments['unassign'] = remove_items
+    res = http_request('POST', 'whitelist/devices', data=assignments)
+    return res
+
+
+def whitelist_retrieve():
+    res_raw = http_request('GET', 'whitelist/devices')
+    res = res_raw.json()
+    return res
+
+
+def add_alert(apply_all, disabled, name, notify_snmp, refire_interval, severity, alert_type, object_type,
+              protocols, field_name, stat_name, units, interval_length, operand, operator, field_name2, field_op,
```

Comment:
`field_name2` and `param2` are a real thing? :suspect:

Reply:
Sadly yes haha